src/SourceGenerators/Uno.UI.SourceGenerators.net6.Tests/Utilities/ISourceGeneratorExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+
+namespace Uno.UI.SourceGenerators.Tests;
+
+internal static class ISourceGeneratorExtensions
+{
+	public static async Task<(GeneratorDriver Driver, ImmutableArray<Diagnostic> Diagnostics)> RunAsync(
+		this ISourceGenerator generator,
+		AnalyzerConfigOptionsProvider options,
+		SyntaxTree[] syntaxTrees,
+		AdditionalText[] additionalTexts,
+		IEnumerable<string>? preprocessorSymbols,
+		CancellationToken cancellationToken = default)
+	{
+		var skiaFolder = Path.Combine("..", "..", "..", "..", "..", "Uno.UI", "bin", "Uno.UI.Skia", "Debug", "netstandard2.0");
+		if (!Directory.Exists(skiaFolder))
+		{
+			throw new InvalidOperationException(
+				"These tests require a project built by Uno.UI.Skia in Debug mode. Please build it using Uno.UI-Skia-only.slnf");
+		}
+
+		var referenceAssemblies = ReferenceAssemblies.Net.Net60;
+		var references = await referenceAssemblies.ResolveAsync(null, cancellationToken);
+		references = references
+			.Add(MetadataReference.CreateFromFile(Path.Combine(skiaFolder, "Uno.dll")))
+			.Add(MetadataReference.CreateFromFile(Path.Combine(skiaFolder, "Uno.Foundation.dll")))
+			.Add(MetadataReference.CreateFromFile(Path.Combine(skiaFolder, "Uno.UI.dll")))
+			.Add(MetadataReference.CreateFromFile(Path.Combine(skiaFolder, "Uno.UI.Composition.dll")));
```

Comment:
This is now the problematic part for CI.

@jeromelaban Do you think there is an issue trying to consume Uno from NuGet instead of locally built binaries? If that's not a good option, I'll try to update the pipeline to build Skia first.